twill-core/src/main/java/org/apache/twill/kafka/client/KafkaConsumer.java

```diff
@@ -33,14 +33,16 @@
 
     /**
      * Invoked when new messages is available.
+     * @param startOffset Offset of the first {@link FetchedMessage} in the iterator of new messages.
```

Comment:
Yes, I think one should be able to use `-1L` to fetch an upcoming new message, which also aligns with the usage of `-1L` in Kafka's `SimpleConsumer`. If the latest message is read, and it's the first message in the iterator, one can just specify `startOffset` as the latest message offset if one doesn't want to skip it. If the latest message is not the first message in the iterator and its previous message was consumed, `onReceived` will still return the latest message offset.